sonaranalyzer-dotnet/src/SonarAnalyzer.Common/SymbolicExecution/AbstractExplodedGraph.cs

```diff
@@ -358,14 +360,24 @@ protected void EnqueueNewNode(ProgramPoint programPoint, ProgramState programSta
                 this.programPoints[pos] = pos;
             }
 
+            ExplodedGraphNode newNode = null;

```

Comment:
MaxProgramPointExecutionCount is currently equal to 2. Did you look at the possibility of increasing it to catch more possible states? other than that the PR looks good to me

Reply:
Increasing it would have an impact on performance, so I believe it's a discussion to have outside this ticket.
I added a test case in the PR to show the limitation (and FP) it can have

Reply:
Great!